sdcm/utils/k8s/__init__.py

```diff
@@ -534,13 +534,11 @@ def gather_k8s_logs_by_operator(cls, kluster, logdir_path=None):
         # Docs: https://github.com/scylladb/scylla-operator/blob/master/docs/source/support/must-gather.md
         logdir_path = logdir_path or kluster.logdir
         kubeconfig = kluster.kube_config_path
-        operator_namespace = kluster._scylla_operator_namespace  # pylint: disable=protected-access
         operator_bin_path = f"/tmp/scylla-operator-{kluster.shortid}"
-        operator_pod_name = kluster.kubectl(
-            "get pods -l app.kubernetes.io/instance=scylla-operator -o custom-columns=:.metadata.name --no-headers",
-            namespace=operator_namespace).stdout.split()[1]
-        kluster.kubectl(
-            f"cp {operator_namespace}/{operator_pod_name}:/usr/bin/scylla-operator {operator_bin_path}")
+        operator_image = kluster.get_operator_image()
+        LOCALRUNNER.run(
+            f"timeout -v 15m docker cp $(docker create --name mustgather {operator_image}):"
+            f"/usr/bin/scylla-operator {operator_bin_path} && docker rm mustgather")
```

Comment:
the whole fact of copying the executable out, is that how the instruction are on using it ?  or it's something we came up with cause of all the credentials issues ?

if it's not in the docs, it might be worse suggesting they update it, it it's a better way of having it running...

Reply:
Their instructions are to use docker image.
Doing so we avoid tricks with kubeconfig cooking.
If it is really needed to go this way, it can be done, we just will have workaround code exactly only for the `must-gather` cooking a kubeconfig for it.